File: valid-anagram.js
var isAnagram = function(s, t) {
    // s and t are strings
    // return true if the string t can be rearranged to form string s
    // Input: s = "anagram", t = "nagaram" Output: true
    // an anagram is defined by one word being able to transform into another word using each original letter once
    // if the length of s and t do not match then return false
    // count the letters of s and t then put the values into their respective objects
    // loop through the object and match the values
    // if any values don't match then return false
    // if the loop ends then return true

    if (s.length !== t.length) return false;

    const sObj = {}, tObj = {};
    for (i = 0; i < s.length; i++) {
        if (sObj[s[i]]) {
            sObj[s[i]]++;
        } else {
            sObj[s[i]] = 1;
        }
    }
    for (i = 0; i < t.length; i++) {
        if (tObj[t[i]]) {
            tObj[t[i]]++;
        } else {
            tObj[t[i]] = 1;
        }
    }
    for (letter in sObj) {
        if (sObj[letter] !== tObj[letter]) return false;
    }
    return true;
};

File: valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # s and t are strings
        # return true if they are anagrams to each other else return false
        # Input: s = "rat", t = "car" Output: false
        # if the lengths of s and t are not equal return false
        # turn s and t into dictionaries that contains the amount of each letters
        # for loop both strings and add the values to the respective dictionary
        # if both dictionaries contain the same amount of letters then they will be considered an anagram
        # for loop through either dictionary and compare key/values
        # if not then return false

        if len(s) != len(t):
            return False

        sDict, tDict = dict(), dict()
        for i in range(len(s)):
            if s[i] in sDict:
                sDict[s[i]] += 1
            else:
                sDict[s[i]] = 1
        for i in range(len(t)):
            if t[i] in tDict:
                tDict[t[i]] += 1
            else:
                tDict[t[i]] = 1

        return sDict == tDict